.github/workflows/on-pr.yml

```diff
@@ -5,6 +5,10 @@ on:
   pull_request:
     branches: [ "main" ]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
There might be cases when someone wants the old workflow to finish in order to compare results with the new commit, but I guess we can manually rerun the workflow in GH Actions tab?

Reply:
Yes, I guess in that case he would need to wait for the workflow to finish before pushing the next change.
Also, you can rerun canceled workflows, and start only the canceled tasks to resume.